browser/BUILD.gn

```diff
@@ -56,6 +56,10 @@ source_set("browser_process") {
     "extensions/brave_component_extension.h",
     "geolocation/brave_geolocation_permission_context.cc",
     "geolocation/brave_geolocation_permission_context.h",
+    "greaselion/greaselion_service_factory.cc",
```

Comment:
I think we probably have enough files here to justify adding a `greaselion/BUILD.gn`. That also helps with tracking dependencies since they don't all get dumped in one place